js/tests/unit/quantity-selector.spec.js

```diff
@@ -0,0 +1,259 @@
+      '    <span class="visually-hidden">Step down</span>',
+      '  </button>',
+      '  <button type="button" class="btn btn-icon btn-secondary" aria-describedby="inputQuantitySelector1" data-bs-step="up">',
+      '    <span class="visually-hidden">Step up</span>',
+      '  </button>',
+      '</div>'
+    ].join('')
+
+    const buttonStepUpEl = fixtureEl.querySelector('[data-bs-step="up"]')
+    const inputEl = fixtureEl.querySelector('[data-bs-step="counter"]')
+
+    expect(inputEl.value).toBe('8.5')
+
+    buttonStepUpEl.click()
+
+    expect(inputEl.value).toBe('9.5')
+
+    expect(buttonStepUpEl.disabled).toBeTruthy()
+  })
+
+  it('should disable click on Step Down button on load to prevent a value out of range', done => {
+    fixtureEl.innerHTML = [
+      '<div class="input-group quantity-selector">',
+      '  <input type="number" id="inputQuantitySelector1" class="form-control" aria-live="polite" data-bs-step="counter" name="quantity" title="quantity" value="0.5" min="0" max="10" step="1" data-bs-round="1" aria-label="Quantity selector">',
+      '  <button type="button" class="btn btn-icon btn-secondary" aria-describedby="inputQuantitySelector1" data-bs-step="down">',
+      '    <span class="visually-hidden">Step down</span>',
+      '  </button>',
+      '  <button type="button" class="btn btn-icon btn-secondary" aria-describedby="inputQuantitySelector1" data-bs-step="up">',
+      '    <span class="visually-hidden">Step up</span>',
+      '  </button>',
+      '</div>'
+    ].join('')
+
+    const buttonStepDownEl = fixtureEl.querySelector('[data-bs-step="down"]')
+    const inputEl = fixtureEl.querySelector('[data-bs-step="counter"]')
+
+    expect(inputEl.value).toBe('0.5')
+
+    buttonStepDownEl.click()
+
+    setTimeout(() => {
+      expect(buttonStepDownEl.disabled).toBeTruthy()
+      done()
+    }, 10)
+  })
+
+  it('should disable click on Step Up button on load to prevent a value out of range', done => {
+    fixtureEl.innerHTML = [
+      '<div class="input-group quantity-selector">',
+      '  <input type="number" id="inputQuantitySelector1" class="form-control" aria-live="polite" data-bs-step="counter" name="quantity" title="quantity" value="9.5" min="0" max="10" step="1" data-bs-round="1" aria-label="Quantity selector">',
+      '  <button type="button" class="btn btn-icon btn-secondary" aria-describedby="inputQuantitySelector1" data-bs-step="down">',
+      '    <span class="visually-hidden">Step down</span>',
+      '  </button>',
+      '  <button type="button" class="btn btn-icon btn-secondary" aria-describedby="inputQuantitySelector1" data-bs-step="up">',
+      '    <span class="visually-hidden">Step up</span>',
+      '  </button>',
+      '</div>'
+    ].join('')
+
+    const buttonStepUpEl = fixtureEl.querySelector('[data-bs-step="up"]')
+    const inputEl = fixtureEl.querySelector('[data-bs-step="counter"]')
+
+    expect(inputEl.value).toBe('9.5')
+
+    buttonStepUpEl.click()
```

Comment:
We shouldn't have to click somewhere or change the focus element to have the disabled button